src/lib/libpthread.js

```diff
@@ -1261,9 +1261,9 @@ var LibraryPThread = {
     }
   }),
 
-  _emscripten_thread_mailbox_await__deps: ['$checkMailbox'],
+  _emscripten_thread_mailbox_await__deps: ['$checkMailbox', '$waitAsyncPolyfilled'],
   _emscripten_thread_mailbox_await: (pthread_ptr) => {
-    if (Atomics.waitAsync) {
+    if (!Atomics.waitAsyncPolyfilled) {
```

Comment:
Why not just `if (waitAsyncPolyfilled)`?  i.e. why modify the Atomics object at all?